src/zenml/integrations/mlflow/experiment_trackers/mlflow_experiment_tracker.py

```diff
@@ -0,0 +1,185 @@
+
+    Manages the global MLflow environment in the form of a Stack
+    component. To access it inside your step function or in the post-execution
+    workflow:
+
+    ```python
+    from zenml.steps import StepContext
+
+    @enable_mlflow
+    @step
+    def my_step(context: StepContext, ...)
+        context.stack.experiment_tracker  # This is the MLFlowExperimentTracker object
+    ```
+    """
+
+    tracking_uri: Optional[str] = None
+    tracking_username: Optional[str] = None
+    tracking_password: Optional[str] = None
+    tracking_token: Optional[str] = None
+    tracking_insecure_tls: bool = False
+
+    # Class Configuration
+    FLAVOR: ClassVar[str] = MLFLOW
+
+    @validator("tracking_uri")
+    def _ensure_valid_tracking_uri(
+        cls, tracking_uri: Optional[str] = None
+    ) -> Optional[str]:
+        """Ensures that the tracking uri is a valid mlflow tracking uri."""
+        if tracking_uri:
+            valid_schemes = DATABASE_ENGINES + ["http", "https", "file"]
+            if not any(
+                tracking_uri.startswith(scheme) for scheme in valid_schemes
+            ):
+                raise ValueError(
+                    f"MLflow tracking uri does not start with one of the valid "
+                    f"schemes {valid_schemes}. See "
+                    f"https://www.mlflow.org/docs/latest/tracking.html#where-runs-are-recorded "
+                    f"for more information."
+                )
+        return tracking_uri
+
+    @root_validator
+    def _ensure_authentication_if_necessary(
+        cls, values: Dict[str, Any]
+    ) -> Dict[str, Any]:
+        """Ensures that credentials or a token for authentication exist when
+        running mlflow tracking with a remote backend."""
+        tracking_uri = values.get("tracking_uri")
+
+        if tracking_uri and any(
+            tracking_uri.startswith(prefix)
+            for prefix in ["http://", "https://"]
+        ):
+            # we need either username + password or a token to authenticate to
+            # the remote backend
+            basic_auth = values.get("tracking_username") and values.get(
+                "tracking_password"
+            )
+            token_auth = values.get("tracking_token")
+
+            if not (basic_auth or token_auth):
+                raise ValueError(
+                    f"MLflow experiment tracking with a remote backend "
+                    f"{tracking_uri} is only when specifying either username "
+                    f"and password or a authentication token in your stack "
+                    f"component. To update your component, run the following "
+                    f"command: `zenml experiment-tracker update {self.name} "
+                    f"--tracking_username=MY_USERNAME "
+                    f"--tracking_password=MY_PASSWORD "
+                    f"--tracking_token=MY_TOKEN` and specify either your "
+                    f"username and password or token."
```

Comment:
This reminds me that we should probably use ZenML secrets to pass the username/password/token instead of embedding them into the stack configuration.